lib/virtfs/iso9660/base.rb

```diff
@@ -0,0 +1,81 @@
+require 'rufus-lru'
+require 'virtfs/nativefs/thin'
+
+module VirtFS::ISO9660
+  class FS
+    attr_accessor :mount_point, :blk_device
+
+    def self.match?(blk_device)
+      blk_device.seek(SUPER_OFFSET + MAGIC_OFFSET)
+      blk_device.read(MAGIC_SIZE) == MAGIC
+    end
+
+    def initialize(blk_device)
+      @blk_device   = blk_device
+      validate_boot_sector!
+    end
+
+    def joliet_boot_sector
+      @joliet_boot_sector ||= begin
+        begin
+          blk_device.seek(JOLIET_SECTOR * SECTOR_SIZE)
+          JolietBootSector.new(blk_device)
+        rescue
+          nil
+        end
+      end
+    end
+
+    def joliet?
+      @joliet ||= !joliet_boot_sector.nil?
+    end
+
+    def primary_boot_sector
+      @primary_boot_sector ||= begin
+        begin
+          blk_device.seek(PRIMARY_SECTOR * SECTOR_SIZE)
+          PrimaryBootSector.new(blk_device)
+        rescue
+          nil
+        end
+      end
+    end
+
+    def primary?
+      @primary ||= !primary_boot_sector.nil?
+    end
+
+    def boot_sector?
+      joliet? || primary?
+    end
+
+    def boot_sector
+      @boot_sector ||= joliet_boot_sector || primary_boot_sector
+    end
+
+    def validate_boot_sector!
+      raise "Couldn't find boot sector" unless boot_sector?
+    end
+
+    def dir_cache
+      @dir_cache ||= LruHash.new(DEF_CACHE_SIZE)
+    end
+
+    def cache_hits
+      @cache_hits ||= 0
+    end
+
+    def drive_root_entry
+      @drive_root_entry ||= DirectoryEntry.new(boot_sector.root_dir_record, boot_sector.suffix)
+    end
+
+    def drive_root
+      @drive_root ||= Directory.new(boot_sector, drive_root_entry)
+    end
+
+    # Wack leading drive leter & colon.
+    def unnormalize_path(p)
+      p[1] == 58 ? p[2, p.size] : p
+    end
+  end # class FS
```

Comment:
@roliveri yes but in the last / cleanup PR which flushed out the odds and ends to make things work end to end

https://github.com/ManageIQ/virtfs-iso9660/pull/10/commits/49283f0490e08dbb028c1b247f2c9a992f424576#diff-2b0d12f2649db8eb256e9aedaa454b64R13